python/activator/middleware_interface.py

```diff
@@ -1781,27 +1785,36 @@ def _filter_datasets(src_repo: Butler,
         Raised if the query on ``src_repo`` failed to find any datasets, or
         (if ``calib_date`` is set) if none of them are currently valid.
     """
-    formatted_args = "{}, {}".format(
+    formatted_args = "dataset_types={}, {}, {}".format(
+        dataset_types,
         ", ".join(repr(a) for a in args),
         ", ".join(f"{k}={v!r}" for k, v in kwargs.items()),
     )
-    try:
-        with lsst.utils.timer.time_this(_log, msg=f"_filter_datasets({formatted_args}) (known datasets)",
-                                        level=logging.DEBUG):
-            known_datasets = set(dest_repo.registry.queryDatasets(*args, **kwargs))
-            _log_trace.debug("Known datasets: %s", known_datasets)
-    except lsst.daf.butler.registry.DataIdValueError as e:
-        _log.debug("Pre-export query with args '%s' failed with %s", formatted_args, e)
-        # If dimensions are invalid, then *any* such datasets are missing.
+    # time_this automatically escalates its log to ERROR level if the code raises.
+    # Some errors are expected to happen sometimes, and we don't want those to log
+    # at the ERROR level and cause confusion.
+    with lsst.utils.timer.time_this(_log, msg=f"_filter_datasets({formatted_args}) (known datasets)",
+                                    level=logging.DEBUG):
         known_datasets = set()
+        for dataset_type in dataset_types:
+            try:
+                subset = set(dest_repo.registry.queryDatasets(dataset_type, *args, **kwargs))
+            except lsst.daf.butler.registry.DataIdValueError as e:
+                _log.debug("Pre-export query with args '%s' failed with %s", formatted_args, e)
+                # If dimensions are invalid, then *any* such datasets are missing.
+            else:
+                known_datasets |= subset
```

Comment:
This seems a bit roundabout, what about:
```suggestion
            try:
                known_datasets |= set(dest_repo.registry.queryDatasets(dataset_type, *args, **kwargs))
            except lsst.daf.butler.registry.DataIdValueError as e:
                _log.debug("Pre-export query with args '%s' failed with %s", formatted_args, e)
                # If dimensions are invalid, then *any* such datasets are missing.
```

Reply:
Corrected. Thanks.